templates/cloudformation/apps/aem/full-set/monitoring.yaml

```diff
@@ -20,409 +20,2065 @@ Parameters:
+      Dimensions:
+        - Name: "AutoScalingGroupName"
+          Value:
+            Fn::ImportValue:
+              Fn::Sub:
+                - "${MainStackPrefixParameter}-PublishAutoScalingGroup"
+                - MainStackPrefixParameter:
+                    Ref: 'MainStackPrefixParameter'
+        - Name: MountPath
+          Value: "/"
+        - Name: Filesystem
+          Value: "/"
+      EvaluationPeriods: 1
+      MetricName: DiskSpaceUtilization
+      Namespace: "System/Linux"
+      OKActions:
+        - Ref: CriticalAlarmNotificationTopic
+      Period: 300
+      Statistic: Average
+      Threshold: 90
+    Type: AWS::CloudWatch::Alarm
+  WarningPublishDataDiskSpaceUtilizationAlarm:
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - Ref: WarningAlarmNotificationTopic
+      AlarmDescription: Disk Space Utilization is high
+      AlarmName:
+        Fn::Sub:
+          - "${MainStackPrefixParameter}-Warning-Publish-DataDiskSpaceHighAlarm"
+          - MainStackPrefixParameter:
+              Ref: 'MainStackPrefixParameter'
+      ComparisonOperator: GreaterThanThreshold
+      Dimensions:
+        - Name: "AutoScalingGroupName"
+          Value:
+            Fn::ImportValue:
+              Fn::Sub:
+                - "${MainStackPrefixParameter}-PublishAutoScalingGroup"
+                - MainStackPrefixParameter:
+                    Ref: 'MainStackPrefixParameter'
+        - Name: MountPath
+          Value: "/mnt/ebs1"
+        - Name: Filesystem
+          Value: "/mnt/ebs1"
+      EvaluationPeriods: 1
+      MetricName: DiskSpaceUtilization
+      Namespace: "System/Linux"
+      OKActions:
+        - Ref: WarningAlarmNotificationTopic
+      Period: 300
+      Statistic: Average
+      Threshold: 80
+    Type: AWS::CloudWatch::Alarm
+  CriticalPublishDataDiskSpaceUtilizationAlarm:
+    Properties:
+      ActionsEnabled: true
+      AlarmActions:
+        - Ref: CriticalAlarmNotificationTopic
+      AlarmDescription: Disk Space Utilization is high
+      AlarmName:
+        Fn::Sub:
+          - "${MainStackPrefixParameter}-Critical-Publish-DataDiskSpaceHighAlarm"
+          - MainStackPrefixParameter:
+              Ref: 'MainStackPrefixParameter'
+      ComparisonOperator: GreaterThanThreshold
+      Dimensions:
+        - Name: "AutoScalingGroupName"
+          Value:
+            Fn::ImportValue:
+              Fn::Sub:
+                - "${MainStackPrefixParameter}-PublishAutoScalingGroup"
+                - MainStackPrefixParameter:
+                    Ref: 'MainStackPrefixParameter'
+        - Name: MountPath
+          Value: "/mnt/ebs1"
+        - Name: Filesystem
+          Value: "/mnt/ebs1"
+      EvaluationPeriods: 1
+      MetricName: DiskSpaceUtilization
+      Namespace: "System/Linux"
+      OKActions:
+        - Ref: CriticalAlarmNotificationTopic
+      Period: 300
+      Statistic: Average
+      Threshold: 90
+    Type: AWS::CloudWatch
```

Comment:
All alarms must be created on the stack which the component lives on.
Adding the alarms on monitoring stack here is inconsistent to the existing alarms.

Reply:
Each alarm moved to its own component stack